Improbable/Stdlib/Improbable.Stdlib/Platform/Platform.cs

```diff
@@ -0,0 +1,236 @@
+            Shell.Run(
+                "spatial", "alpha", "config", "convert", "--json_output", "--launch_config", Shell.Escape(startDeployment.LaunchConfigPath),
+                "--main_config", Shell.Escape(startDeployment.ProjectConfigPath), "--output_path", splConfigPath);
+
+            var project = JsonConvert.DeserializeObject<ProjectConfig>(File.ReadAllText(startDeployment.ProjectConfigPath, Encoding.UTF8));
+
+            var needsStart = true;
+            var pidFile = Path.Combine(ToolbeltConfigDir, "spatiald.pid");
+            var lockFile = Path.Combine(ToolbeltConfigDir, "spatiald.lock");
+            if (File.Exists(pidFile) && int.TryParse(File.ReadAllText(pidFile).Trim(), out var pid))
+            {
+                try
+                {
+                    Process.GetProcessById(pid);
+                    needsStart = false;
+                }
+                catch
+                {
+                    File.Delete(pidFile);
+
+                    if (File.Exists(lockFile))
+                    {
+                        File.Delete(lockFile);
+                    }
+                }
+            }
+
+            if (needsStart)
+            {
+                Shell.Run("spatial", "service", "start", "--json_output", "--main_config", Shell.Escape(startDeployment.ProjectConfigPath));
+            }
+
+            using (var tcs = new CancellationTokenSource(5000))
+            {
+                var timeout = CallSettings.FromCancellationToken(tcs.Token);
+                var deployments = client.ListDeployments(new ListDeploymentsRequest
+                    { DeploymentName = "local", ProjectName = project.ProjectName }, timeout);
+
+                foreach (var dpl in deployments.Where(dpl => dpl.Status == Deployment.Types.Status.Running))
+                {
+                    client.DeleteDeployment(new DeleteDeploymentRequest { Id = dpl.Id });
+                }
+            }
+
+            var response = client.CreateDeployment(new CreateDeploymentRequest
+            {
+                Deployment = new Deployment
+                {
+                    Tag = { startDeployment.Tags },
+                    Name = "local",
+                    StartingSnapshotId = startDeployment.SnapshotId,
+                    ProjectName = project.ProjectName,
+                    LaunchConfig = new SpatialOS.Deployment.V1Alpha1.LaunchConfig
+                    {
+                        ConfigJson = File.ReadAllText(splConfigPath)
+                    }
+                }
+            }).PollUntilCompleted().Result;
+
+            // Apply desired starting worker flags.
+            if (startDeployment.WorkerFlags.Any())
+            {
+                var allWorkerFlags = new RepeatedField<WorkerFlag>();
+
+                var startFlags = startDeployment.WorkerFlags.SelectMany(wf =>
+                    wf.Flags.Select(f => new WorkerFlag { Key = f.Name, Value = f.Value, WorkerType = wf.WorkerType }));
```

Comment:
Worth providing interop between on the `Platform.DeploymentWorkerFlags` object and this one to tidy this up? 

I.e. - 

```csharp
public IEnumerable<WorkerFlag> ToPlatformSdk()
{
	...
}
```